includes/class-events-store.php

```diff
@@ -0,0 +1,592 @@
+	 *
+	 * @param  array $attrs Array of event attributes to query by
+	 * @return object|false
+	 */
+	public function get_job( $attrs ) {
+		global $wpdb;
+
+		// Validate basic inputs
+		if ( ! is_array( $attrs ) || empty( $attrs ) ) {
+			return false;
+		}
+
+		// Validate requested status
+		$allowed_status = $this->get_allowed_statuses();
+		$allowed_status[] = 'any';
+
+		if ( ! isset( $attrs['status'] ) || ! in_array( $attrs['status'], $allowed_status, true ) ) {
+			$attrs['status'] = self::STATUS_PENDING;
+		}
+
+		// Validate attributes provided to query for a post
+		$query              = null;
+		$query_replacements = array();
+
+		if ( isset( $attrs['ID'] ) ) {
+			$query                = "SELECT * FROM {$this->get_table_name()} WHERE ID = %d";
+			$query_replacements[] = $attrs['ID'];
+		} else {
+			// Need a timestamp, an instance, and either an action or its hashed representation
+			if ( ! isset( $attrs['timestamp'] ) || ! isset( $attrs['instance'] ) ) {
+				return false;
+			} elseif ( ! isset( $attrs['action'] ) && ! isset( $attrs['action_hashed'] ) ) {
+				return false;
+			}
+
+			// Build query
+			if ( isset( $attrs['action'] ) ) {
+				$action_column = 'action';
+				$action_value  = $attrs['action'];
+			} else {
+				$action_column = 'action_hashed';
+				$action_value  = $attrs['action_hashed'];
+			}
+
+			// Do not sort, otherwise index isn't used
+			$query                = "SELECT * FROM {$this->get_table_name()} WHERE timestamp = %d AND {$action_column} = %s AND instance = %s";
+			$query_replacements[] = $attrs['timestamp'];
+			$query_replacements[] = $action_value;
+			$query_replacements[] = $attrs['instance'];
+		}
+
+		// Final query preparations
+		if ( 'any' !== $attrs['status'] ) {
+			$query                .= ' AND status = %s';
+			$query_replacements[]  = $attrs['status'];
+		}
+
+		$query .= ' LIMIT 1';
+
+		$query = $wpdb->prepare( $query, $query_replacements );
+
+		// Query and format results
+		$job = $wpdb->get_row( $query );
+
+		if ( is_object( $job ) && ! is_wp_error( $job ) ) {
+			$job = $this->format_job( $job );
+		} else {
+			$job = false;
+		}
+
+		return $job;
+	}
+
+	/**
+	 * Standardize formatting and expand serialized data
+	 *
+	 * @param  object $job Job row from DB, in object form
+	 * @return object
+	 */
+	private function format_job( $job ) {
+		if ( ! is_object( $job ) || is_wp_error( $job ) ) {
+			return $job;
+		}
+
+		$job->ID        = (int) $job->ID;
+		$job->timestamp = (int) $job->timestamp;
+		$job->interval  = (int) $job->interval;
+		$job->args      = maybe_unserialize( $job->args );
+
+		if ( empty( $job->schedule ) ) {
+			$job->schedule = false;
+		}
+
+		return $job;
+	}
+
+	/**
+	 * Check if a job post exists
+	 *
+	 * Uses a direct query to avoid stale caches that result in duplicate events
+	 */
+	public function job_exists( $timestamp, $action, $instance, $return_id = false ) {
```

Comment:
This is one of several vestigial patterns in this plugin, which I'm not thrilled about. I'll look at fixing this.